{
  "comments": [
    {
      "key": {
        "uuid": "ba068154_980c8744",
        "filename": "csit/libraries/controller/DdbCommons.robot",
        "patchSetId": 3
      },
      "lineNbr": 188,
      "author": {
        "id": 2046
      },
      "writtenOn": "2017-05-15T11:46:15Z",
      "side": 1,
      "message": "Line 133 has started 3 writers, each is expected to be failed by now. This removes one response, other ones will be extracted in line 196, causing a failure in line 198.\n\nEither extract failures here, or perhaps reset globals in line 195.",
      "range": {
        "startLine": 188,
        "startChar": 33,
        "endLine": 188,
        "endChar": 63
      },
      "revId": "a41bc9031102a287021b9cd48968787ac1d5ec0c",
      "serverId": "7fc14799-209e-464c-9743-7a06c2c21a81",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "ba068154_13bb58e6",
        "filename": "csit/libraries/controller/DdbCommons.robot",
        "patchSetId": 3
      },
      "lineNbr": 188,
      "author": {
        "id": 2630
      },
      "writtenOn": "2017-05-15T12:16:39Z",
      "side": 1,
      "message": "Wait_For_Transactions does not really care about the number of responses, so it will not fail on 198. It cares about the threads/http requests to finish. It returns all responses present in the queue from threads.\n\nWait_For_Transactions returns the first response from the queue, leaving thread object untouched.\n\nThe behavior described will not happen.",
      "parentUuid": "ba068154_980c8744",
      "range": {
        "startLine": 188,
        "startChar": 33,
        "endLine": 188,
        "endChar": 63
      },
      "revId": "a41bc9031102a287021b9cd48968787ac1d5ec0c",
      "serverId": "7fc14799-209e-464c-9743-7a06c2c21a81",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "ba068154_a517a3ec",
        "filename": "csit/libraries/controller/DdbCommons.robot",
        "patchSetId": 3
      },
      "lineNbr": 188,
      "author": {
        "id": 2046
      },
      "writtenOn": "2017-05-15T15:01:35Z",
      "side": 1,
      "message": "Here [0] I see multiple responses. I was only expecting one, from the writer at line 194. So I suspect the others are from failed writers at line 159.\n\n[0] https://logs.opendaylight.org/releng/jenkins092/controller-csit-3node-clustering-only-carbon/684/archives/log.html.gz#s1-s33-t2-k2-k23-k1-k7",
      "parentUuid": "ba068154_13bb58e6",
      "range": {
        "startLine": 188,
        "startChar": 33,
        "endLine": 188,
        "endChar": 63
      },
      "revId": "a41bc9031102a287021b9cd48968787ac1d5ec0c",
      "serverId": "7fc14799-209e-464c-9743-7a06c2c21a81",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "ba068154_05bb0fe6",
        "filename": "csit/libraries/controller/DdbCommons.robot",
        "patchSetId": 3
      },
      "lineNbr": 188,
      "author": {
        "id": 2630
      },
      "writtenOn": "2017-05-15T15:11:12Z",
      "side": 1,
      "message": "Not really.\n3 producers started [1]. Then one node is isolated. Get_Next_Transaction_Response at line 188 received the response from that node.\nProducer on this node is restarted again(line 194), so again there are 3 active producers.\nAnd their answers are received at 198 [0].\n\n\n[1] https://logs.opendaylight.org/releng/jenkins092/controller-csit-3node-clustering-only-carbon/684/archives/log.html.gz#s1-s33-t2-k2-k6.",
      "parentUuid": "ba068154_a517a3ec",
      "range": {
        "startLine": 188,
        "startChar": 33,
        "endLine": 188,
        "endChar": 63
      },
      "revId": "a41bc9031102a287021b9cd48968787ac1d5ec0c",
      "serverId": "7fc14799-209e-464c-9743-7a06c2c21a81",
      "unresolved": false
    }
  ]
}